### PR TITLE
[SmartAccount] Get authenticator query

### PR DESCRIPTION
**Pull request by iboss-ptk**
## What is the purpose of the change

Query for getting single authenticator + stargate whitelist

## Testing and Verifying

Tested through integration spend limit contract

## Summary by CodeRabbit

- **New Features**
	- Introduced a new feature to fetch specific authenticators for smart accounts, enhancing user capabilities in managing account securities.

- **Enhancements**
	- Expanded whitelisted queries to support new smart account functionalities, improving interaction security and compliance.

- **Bug Fixes**
	- Updated error message formatting in spend limit tests to include more precise time details, enhancing debugging accuracy.

### CODE REVIEW

File: x/smart-account/authenticator/spend_limits_test.go [hunk 1]
```diff
@@ -281,7 +281,7 @@ func (s *SpendLimitAuthenticatorTest) TestSpendLimit() {
 	s.Require().Contains(
 		err.Error(),
 		fmt.Sprintf(
-			"Current time %d.%d not within time limit None - %s.%s: execute wasm contract failed",
+			"Current time %d.%09d not within time limit None - %s.%s: execute wasm contract failed",
```
Comment (nicolaslara) [inline x/smart-account/authenticator/spend_limits_test.go:284]: wasn't this already merged? https://github.com/osmosis-labs/osmosis/pull/8120/files

**Review by nicolaslara — APPROVED** [review]

**Review by PaddyMc — APPROVED** [review]

**Review by PaddyMc — COMMENTED** [review]
Would you mind adding a CHANGELOG entry?

File: x/smart-account/authenticator/spend_limits_test.go (hunk 1 above)
Comment (iboss-ptk) [reply x/smart-account/authenticator/spend_limits_test.go:284]: yeah not sure why it's in here.

Comment (iboss-ptk) [reply x/smart-account/authenticator/spend_limits_test.go:284]: will revert the commit and let the merge figure out.

**Comment by iboss-ptk** [comment]
> Would you mind adding a CHANGELOG entry?

Added

**Pull request merged**